base_layer/wallet/src/output_manager_service/storage/sqlite_db/output_sql.rs

```diff
@@ -483,10 +484,12 @@ impl TryFrom<OutputSql> for DbUnblindedOutput {
     type Error = OutputManagerStorageError;
 
     fn try_from(o: OutputSql) -> Result<Self, Self::Error> {
-        let mut features: OutputFeatures =
-            serde_json::from_str(o.features_json.as_str()).map_err(|s| OutputManagerStorageError::ConversionError {
-                reason: format!("Could not convert json into OutputFeatures:{}", s),
-            })?;
+        let mut features: OutputFeatures = serde_json::from_str(
+            &OutputFeatures::add_recovery_byte_to_serialized_data_if_needed(o.features_json.clone()),
```

Comment:
This should only be done at wallet startup, not every time transaction data is read from the db.